pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -87,10 +88,18 @@ public MultiStageBrokerRequestHandler(PinotConfiguration config, BrokerRoutingMa
     }
     _reducerHostname = reducerHostname;
     _reducerPort = config.getProperty(QueryConfig.KEY_OF_QUERY_RUNNER_PORT, QueryConfig.DEFAULT_QUERY_RUNNER_PORT);
+    long brokerTimeoutMs = config.getProperty(CommonConstants.Broker.CONFIG_OF_BROKER_TIMEOUT_MS, -1L);
+    if (brokerTimeoutMs == -1) {
+      _timeoutNs = config.getProperty(QueryConfig.KEY_OF_QUERY_RUNNER_TIMEOUT_NANO,
+          QueryConfig.DEFAULT_QUERY_RUNNER_TIMEOUT_NANO);
+    } else {
+      _timeoutNs = brokerTimeoutMs * 1_000_000L;
```

Comment:
nit: can we use `TimeUnit.MILLIS.toNanos(brokerTimeoutMs)` - makes it a bit easier to read.